File: set_1/easy/javascript/feb1f421-6645-4402-bc05-051d4dde2178.js
// Solution 1:
const removeDupes = arr => {
    let tempArr = [];
    arr.forEach(element => {
        if (!tempArr.includes(element)) {
            tempArr.push(element)
        }
    });
    return tempArr;
}

const findNLargest = (arr, n) => {
    let uniqueArr = removeDupes(arr)
    let sortedArr = uniqueArr.sort((el1, el2) => {
        return el1 - el2
    });
    console.log(`Sorted unique array --> [${sortedArr}]`)
    let sortedArrLen = sortedArr.length
    return n === 1 ? sortedArr[sortedArrLen - 1] : sortedArr[sortedArrLen - n]

}

const arr1 = [407, 248, 169, 232, 371, 470, 86, 209, 465, 193]
const n = 3
console.log(`The input array --> [${arr1}]`);
const res1 = findNLargest(arr1, n);
console.log(`The ${n}th largest number is --> ${res1}`);


// Solution 2:
const evenOddRearranger2 = arr => {
    let arrLength = arr.length;
    let [i,j] = [-1,0];
    while (j != arrLength) {
        if (arr[j] % 2 == 0) {
            i++;
            let temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
        j++;
    }

    return arr
}

const arr2 = [9, 18, 23, 28, 31, 32, 35, 38, 48, 49];
console.log(`The input array --> [${arr2}]`)
const res2 = evenOddRearranger2(arr2);
console.log(`The array after even-odd re-arrangement --> [${res2}]`);

File: set_1/easy/python/feb1f421-6645-4402-bc05-051d4dde2178.py
# Solution 1:
def remove_dupes(arr):
    return list(set(arr))

def find_n_largest(arr, n):
    unique_arr = remove_dupes(arr)
    unique_arr.sort()
    unique_arr_len = len(unique_arr)
    print(f"Unique sorted array --> {unique_arr}")
    
    return unique_arr[unique_arr_len-1] if n==1 else unique_arr[unique_arr_len-n]

arr1 = [407, 248, 169, 232, 371, 470, 86, 209, 465, 193]
n=5
print(f"the input array --> {arr1}")
res1 = find_n_largest(arr1, n)
print(f"The {n}th largest number is --> {res1}")
